### DISCUSSION TITLE
How to limit the number of connections？

### DISCUSSION THREAD

**Discussion by cutiechi** [apache/kyuubi#2288] [Q&A] [answered]
I want to limit the number of connections

**Comment by pan3793** [comment, marked as answer]
The max connections of thrift server is contronlled by `kyuubi.frontend.thrift.max.worker.threads`
https://kyuubi.apache.org/docs/latest/deployment/settings.html#kyuubi-configurations

**Reply by cutiechi** [reply]
Sorry, I mean that each user sets it separately

**Reply by pan3793** [reply]
I don't get your point, could you explain more about what's "it" mean?

**Reply by cutiechi** [reply]
Each user corresponds to an engine, and I want to set different connection limit for these users

**Reply by pan3793** [reply]
try
```
___tom___.kyuubi.frontend.thrift.max.worker.threads
___bob___.kyuubi.frontend.thrift.max.worker.threads
```
it only effect the user's corresponding engine

**Reply by cutiechi** [reply]
ok

**Comment by wForget** [comment]
Do you mean to need a configuration like `hive.server2.limit.connections.per.user`?
Kyuubi does not support it yet, maybe we can implement it.

**Reply by yaooqinn** [reply]
+1

**Reply by wForget** [reply]
I have created issue #2301 and I will try to implement it.
